Stream browser nasty looking error message

Under System Stream browser set query to `Status in 1,2,3,4`

Get nasty popup containing 

```
ResponseException{method='POST', url='https://xxxxxxxxx/api/meta/v1/find', json='{"code":500,"message":"No enum constant stroom.meta.shared.Status.1","details":"java.lang.IllegalArgumentException: No enum constant stroom.meta.shared.Status.1"}', code='500', message='No enum constant stroom.meta.shared.Status.1', details='java.lang.IllegalArgumentException: No enum constant stroom.meta.shared.Status.1', responseKeyValues='code: 500, message: No enum constant stroom.meta.shared.Status.1, details: java.lang.IllegalArgumentException: No enum constant stroom.meta.shared.Status.1'}
	at Unknown.Hh(stroom-0.js)
	at Unknown.Uh(stroom-0.js)
	at Unknown.new $nf(stroom-0.js)
	at Unknown.Hnf(stroom-0.js)
	at Unknown.Onf(stroom-0.js)
	at Unknown.kod(stroom-0.js)
	at Unknown.dw(stroom-0.js)
	at Unknown.rw(stroom-0.js)
	at Unknown.eval(stroom-0.js)
	at Unknown.ij(stroom-0.js)
	at Unknown.lj(stroom-0.js)
	at Unknown.eval(stroom-0.js)
Caused by: org.fusesource.restygwt.client.FailedResponseException: 
	at Unknown.Rh(stroom-0.js)
	at Unknown.Th(stroom-0.js)
	at Unknown.rod(stroom-0.js)
	at Unknown.new sod(stroom-0.js)
	at Unknown.kod(stroom-0.js)
	at Unknown.dw(stroom-0.js)
	at Unknown.rw(stroom-0.js)
	at Unknown.eval(stroom-0.js)
	at Unknown.ij(stroom-0.js)
	at Unknown.lj(stroom-0.js)
	at Unknown.eval(stroom-0.js)
````

--- a/stroom-core-client/src/main/java/stroom/alert/client/event/AlertEvent.java
+++ b/stroom-core-client/src/main/java/stroom/alert/client/event/AlertEvent.java
@@ -30,30 +30,42 @@ public class AlertEvent extends CommonAlertEvent<AlertEvent.Handler> {
     public static GwtEvent.Type<Handler> TYPE;
     private final AlertCallback callback;
 
-    private AlertEvent(final SafeHtml message, final SafeHtml detail, final Level level, final AlertCallback callback) {
+    private AlertEvent(final SafeHtml message,
+                       final SafeHtml detail,
+                       final Level level,
+                       final AlertCallback callback) {
         super(message, detail, level);
         this.callback = callback;
     }
 
-    public static void fireInfo(final HasHandlers handlers, final String message, final String detail,
+    public static void fireInfo(final HasHandlers handlers,
+                                final String message,
+                                final String detail,
                                 final AlertCallback callback) {
         handlers.fireEvent(new AlertEvent(fromString(message), fromString(detail), Level.INFO, callback));
     }
 
-    public static void fireInfo(final HasHandlers handlers, final String message, final AlertCallback callback) {
+    public static void fireInfo(final HasHandlers handlers,
+                                final String message,
+                                final AlertCallback callback) {
         fireInfo(handlers, message, null, callback);
     }
 
-    public static void fireWarn(final HasHandlers handlers, final String message, final String detail,
+    public static void fireWarn(final HasHandlers handlers,
+                                final String message,
+                                final String detail,
                                 final AlertCallback callback) {
         handlers.fireEvent(new AlertEvent(fromString(message), fromString(detail), Level.WARN, callback));
     }
 
-    public static void fireWarn(final HasHandlers handlers, final String message, final AlertCallback callback) {
+    public static void fireWarn(final HasHandlers handlers,
+                                final String message,
+                                final AlertCallback callback) {
         fireWarn(handlers, message, null, callback);
     }
 
-    public static void fireErrorFromException(final HasHandlers handlers, final Throwable throwable,
+    public static void fireErrorFromException(final HasHandlers handlers,
+                                              final Throwable throwable,
                                               final AlertCallback callback) {
         String message = throwable.getMessage();
         if (message == null || message.length() <= 1) {
@@ -62,8 +74,10 @@ public class AlertEvent extends CommonAlertEvent<AlertEvent.Handler> {
         fireErrorFromException(handlers, message, throwable, callback);
     }
 
-    public static void fireErrorFromException(final HasHandlers handlers, final String message,
-                                              final Throwable throwable, final AlertCallback callback) {
+    public static void fireErrorFromException(final HasHandlers handlers,
+                                              final String message,
+                                              final Throwable throwable,
+                                              final AlertCallback callback) {
         final StringBuilder detail = new StringBuilder();
 
         if (throwable instanceof EntityServiceException) {
@@ -90,11 +104,15 @@ public class AlertEvent extends CommonAlertEvent<AlertEvent.Handler> {
         fireError(handlers, message, detail.toString(), callback);
     }
 
-    public static void fireError(final HasHandlers handlers, final String message, final AlertCallback callback) {
+    public static void fireError(final HasHandlers handlers,
+                                 final String message,
+                                 final AlertCallback callback) {
         fireError(handlers, message, null, callback);
     }
 
-    public static void fireError(final HasHandlers handlers, final String message, final String detail,
+    public static void fireError(final HasHandlers handlers,
+                                 final String message,
+                                 final String detail,
                                  final AlertCallback callback) {
         handlers.fireEvent(new AlertEvent(fromString(message), fromString(detail), Level.ERROR, callback));
     }
@@ -119,6 +137,10 @@ public class AlertEvent extends CommonAlertEvent<AlertEvent.Handler> {
     public AlertCallback getCallback() {
         return callback;
     }
+
+
+    // --------------------------------------------------------------------------------
+
 
     public interface Handler extends EventHandler {
 

--- a/stroom-core-client/src/main/java/stroom/alert/client/view/CommonAlertViewImpl.java
+++ b/stroom-core-client/src/main/java/stroom/alert/client/view/CommonAlertViewImpl.java
@@ -22,6 +22,7 @@ import stroom.widget.util.client.MouseUtil;
 import stroom.widget.util.client.SvgImageUtil;
 
 import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.regexp.shared.RegExp;
 import com.google.gwt.safehtml.shared.SafeHtml;
 import com.google.gwt.safehtml.shared.SafeHtmlBuilder;
 import com.google.gwt.user.client.ui.FlowPanel;
@@ -35,6 +36,8 @@ import com.gwtplatform.mvp.client.ViewImpl;
 public class CommonAlertViewImpl extends ViewImpl implements CommonAlertView {
 
     private static final int MAX_MESSAGE_LENGTH = 1000;
+    // Regex is testing html, hence the <br>
+    private static RegExp STACK_TRACE_PATTERN = RegExp.compile("<br>(\\sat |Caused by: )");
 
     private final FlowPanel layout = new FlowPanel();
     private final SimplePanel image = new SimplePanel();
@@ -120,7 +123,9 @@ public class CommonAlertViewImpl extends ViewImpl implements CommonAlertView {
             builder.append(html);
             builder.appendHtmlConstant("</div>");
             detail.getElement().setInnerHTML(builder.toSafeHtml().asString());
-            showDetail(true);
+            // If it looks like a stack trace then your average user doesn't want to see that nastiness,
+            // but if not then it is probably useful for them to see.
+            showDetail(!STACK_TRACE_PATTERN.test(html.asString()));
         } else {
             showHideDetail.setVisible(false);
             setDetailVisible(false);

--- a/stroom-meta/stroom-meta-impl-db/src/main/java/stroom/meta/impl/db/MetaDaoImpl.java
+++ b/stroom-meta/stroom-meta-impl-db/src/main/java/stroom/meta/impl/db/MetaDaoImpl.java
@@ -227,7 +227,7 @@ public class MetaDaoImpl implements MetaDao {
                 MetaFields.PIPELINE_NAME, metaProcessor.PIPELINE_UUID, this::getPipelineUuidsByName, true);
         expressionMapper.map(MetaFields.STATUS,
                 meta.STATUS,
-                value -> MetaStatusId.getPrimitiveValue(Status.valueOf(value.toUpperCase())));
+                value -> MetaStatusId.getPrimitiveValue(value.toUpperCase()));
         expressionMapper.map(MetaFields.STATUS_TIME, meta.STATUS_TIME, value -> getDate(MetaFields.STATUS_TIME, value));
         expressionMapper.map(MetaFields.CREATE_TIME, meta.CREATE_TIME, value -> getDate(MetaFields.CREATE_TIME, value));
         expressionMapper.map(MetaFields.EFFECTIVE_TIME,
@@ -238,7 +238,7 @@ public class MetaDaoImpl implements MetaDao {
         expressionMapper.map(MetaFields.PARENT_ID, meta.PARENT_ID, Long::valueOf);
         expressionMapper.map(MetaFields.PARENT_STATUS,
                 parent.STATUS,
-                value -> MetaStatusId.getPrimitiveValue(Status.valueOf(value.toUpperCase())));
+                value -> MetaStatusId.getPrimitiveValue(value.toUpperCase()));
         expressionMapper.map(MetaFields.PARENT_CREATE_TIME,
                 parent.CREATE_TIME,
                 value -> getDate(MetaFields.PARENT_CREATE_TIME, value));

--- a/stroom-meta/stroom-meta-impl-db/src/main/java/stroom/meta/impl/db/MetaStatusId.java
+++ b/stroom-meta/stroom-meta-impl-db/src/main/java/stroom/meta/impl/db/MetaStatusId.java
@@ -17,6 +17,12 @@
 package stroom.meta.impl.db;
 
 import stroom.meta.shared.Status;
+import stroom.util.NullSafe;
+import stroom.util.logging.LogUtil;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.stream.Collectors;
 
 /**
  * <p>
@@ -36,6 +42,22 @@ class MetaStatusId {
             case LOCKED -> 1;
             case DELETED -> 99;
         };
+    }
+
+    static byte getPrimitiveValue(final String statusName) {
+        if (NullSafe.isBlankString(statusName)) {
+            throw new IllegalArgumentException("No status name supplied");
+        }
+        try {
+            final Status status = Status.valueOf(statusName.toUpperCase());
+            return getPrimitiveValue(status);
+        } catch (IllegalArgumentException e) {
+            throw new RuntimeException(LogUtil.message("'{}' is not a valid status value. Valid values are: '{}'",
+                    statusName,
+                    Arrays.stream(Status.values())
+                            .map(Objects::toString)
+                            .collect(Collectors.joining(", "))), e);
+        }
     }
 
     static Status getStatus(byte primitiveValue) {

--- a/unreleased_changes/20231004_105657_832__3745.md
+++ b/unreleased_changes/20231004_105657_832__3745.md
@@ -1,0 +1,24 @@
+* Issue **#3745** : Improve error message when filtering using an invalid stream Status value.
+
+
+```sh
+# ********************************************************************************
+# Issue title: Stream browser nasty looking error message
+# Issue link:  https://github.com/gchq/stroom/issues/3745
+# ********************************************************************************
+
+# ONLY the top line will be included as a change entry in the CHANGELOG.
+# The entry should be in GitHub flavour markdown and should be written on a SINGLE
+# line with no hard breaks. You can have multiple change files for a single GitHub issue.
+# The  entry should be written in the imperative mood, i.e. 'Fix nasty bug' rather than
+# 'Fixed nasty bug'.
+#
+# Examples of acceptable entries are:
+#
+#
+# * Issue **123** : Fix bug with an associated GitHub issue in this repository
+#
+# * Issue **namespace/other-repo#456** : Fix bug with an associated GitHub issue in another repository
+#
+# * Fix bug with no associated GitHub issue.
+```

--- a/unreleased_changes/20231004_105735_192__0.md
+++ b/unreleased_changes/20231004_105735_192__0.md
@@ -1,0 +1,19 @@
+* Change alert dialogs to not show the detail by default if the detail looks like a stack trace.
+
+
+```sh
+# ONLY the top line will be included as a change entry in the CHANGELOG.
+# The entry should be in GitHub flavour markdown and should be written on a SINGLE
+# line with no hard breaks. You can have multiple change files for a single GitHub issue.
+# The  entry should be written in the imperative mood, i.e. 'Fix nasty bug' rather than
+# 'Fixed nasty bug'.
+#
+# Examples of acceptable entries are:
+#
+#
+# * Issue **123** : Fix bug with an associated GitHub issue in this repository
+#
+# * Issue **namespace/other-repo#456** : Fix bug with an associated GitHub issue in another repository
+#
+# * Fix bug with no associated GitHub issue.
+```

